pkg/registry/ecr.go

```diff
@@ -0,0 +1,107 @@
+package registry
```

Comment:
move to `pkg/registry/ecr/ecr.go`

make sure its struct uses the abstract as the abstract will have the common functionallity between registries